### DISCUSSION TITLE
Labels for isolines between isobands

### DISCUSSION THREAD

**Discussion by rd-atmosphericg2** [Turfjs/turf#2404] [Q&A]
Hello, I am using turfjs to plot isobands of temperature on a map. I'd like to to add a label to the isolines that are drawn between the isobands - the label would be the temperature along that isoline. Is that something that turfjs can do?

Thank You,
Roy

**Comment by twelch** [comment]
Hi @rd-atmosphericg2, turf is not exactly involved with creating end maps or things related to presentation on maps like labeling.  It creates GeoJSON data and you have to figure out how to plug it in.  Take a look at the features output by the call to the isobands function and see if one of its properties contain text appropriate to use for labeling, and figure out how your mapping client can reference that property for a label.

**Reply by twelch** [reply]
For example if I were using Mapbox or ArcGIS online tooling, I would use its built-in support for creating a label layer that reference a property of my lines, and offsetting that label by a set amount and direction from the line.

**Reply by rd-atmosphericg2** [reply]
So create a new layer that places temperature labels close to the coordinates of each MultiPolygon (isoband) that is returned by the isobands function. The problem with that is the MultiPolygon array in each feature is empty. The other piece of information that would be needed, the temperature range in the band, is present in the feature properties.

**Reply by rd-atmosphericg2** [reply]
Thank you for your suggestion, @twelch. I think it would allow for labels to be added by the caller of the isobands method. I added an issue for the coordinates not being present in the features returned by the isobands method: https://github.com/Turfjs/turf/issues/2405

**Comment by smallsaucepan** [comment]
Resolved related issue as hadn't received any further feedback, so resolving this discussion as well. If some further info becomes available please add it here.
